Task estimation topic (#8)
Alp4ga/SQA_Assignment_1: Handbook.md

### 2.2 Important guidelines
(Write in bullet points)
### 2.3 Topics
One of the hardest question you could ask to a developer would probably be "how much time do you need to complete the following task ?".
There are lots of differents systems out there, some use points, other system directly use time units as a man-days.
You will find yourself in the situation of estimating a task you never done before, so you'll have to take in consideration the learning time and possibles issues you will encounter during the development phases.
Since you cannot just pick a skill and multiply it by the amount of times it is required and then sell it to your boss or your client, you have to find a way to estimate every task and minimizing the margin of error.

The idea is that the more time you spend on the estimation, the more accurate it will be. But then, if you spend too much time on it, then it will be more expensive than a poorly accurate estimation. We have to find the right balance on time estimation, practice is the key here.

> When it comes to the technic and how to foresee the time estimation, it pretty much always come down to experience.

Reviewer: ```suggestion
When it comes to the technique and how to foresee the time estimation, it pretty much always come down to experience.
```

First solution is to use your own experience, try to keep track of the time spent on similar tasks so you can use this data to be the most accurate possible in the future.
In the case of a new technology, or a new technology stack, don't take the risk, take time to identify the problems and study them correctly, doing a POC could also be a solution.
Another way to solve this problem, or a way to be more accurate, is to ask an expert or a colleague, so he will be able to help you and comment on your decisions.
Every feedback is good to hear during this phase of the planning.

> It will help you find an smart solution and estimate your time accurately.

Reviewer: ```suggestion
It will help you find a smart solution and estimate your time accurately.
```

When your tasks are estimated, you will have to write it down and save it so everybody in the team will be able to have the same information about this matter.

> Very differents solutions exist in this regard, it ranges from simple Kanban board tools to a very complex cluster of various tools in order to have a complete overview of the task, the time estimation, the progress and the differents unforeseen issues that the support service will give as feedback for us.

Reviewer: ```suggestion
Very different solutions exist in this regard, from simple Kanban board tools to a very complex cluster of various tools in order to have a complete overview of the task, the time estimation, the progress and the different unforeseen issues that the support service will give as feedback for us.
```

The use of Kanban board can be very helpful when it comes to plan the different issues and sort them by relevance and priority.
But it lacks a real time management module and make the presentation of the data very hard to overview for the manager.

> That's why other complex solutions are available for bigger project.

Reviewer: ```suggestion
That's why other complex solutions are available for bigger projects.
```

> If you're project is simple, and doesn't need a long-time support, a simple Kanban board as Trello of the board in GitLab tool is enough.

Reviewer: ```suggestion
If your project is simple, and does not need a long-time support, a simple Kanban board as Trello of the board in GitLab tool is enough.
```

> On the other hand, if your solution needs a better overview of the multiple tasks and have also a crucial need of support in the long run, more complex tools as Jira should be considered.

Reviewer: ```suggestion
On the other hand, if your solution needs a better overview of the multiple tasks and also have a crucial need of support in the long run, more complex tools as Jira should be considered.
```
